### ISSUE TITLE
Memory leak when using LinePlot.set_data

### ISSUE THREAD

**Issue by UUSim** [vispy/vispy#1407]
I've modified the example [plot.py](https://github.com/vispy/vispy/blob/master/examples/basics/plotting/plot.py) to continuously update the plot data.
To achieve this, I'm using the `set_data` function on the `line` object (`LinePlot`), as follows:

```
N_SAMPLES = 1000000
x = np.linspace(0, 1, N_SAMPLES)
y = np.random.rand(N_SAMPLES)-.5
line.set_data((x,y), symbol=None)
```

From a functional perspective this works, i.e. the new data is being displayed, however, this results in a huge memory leak (approx. 57 MiB leaked per call to `set_data`).

I'm using vispy `0.5.2` and `python3.5.2` on linux.

I am aware of the [realtime_signals.py](https://github.com/vispy/vispy/blob/master/examples/demo/gloo/realtime_signals.py) example, which has updating data. However, this example doesn't have axes, etc.

How can I achieve a plot with updating data, without memory leaks, and with plot labels, axes, etc?

**Comment by davidh-ssec** [comment]
> How can I achieve a plot with updating data, without memory leaks, and with plot labels, axes, etc?

The answer is simple: pull requests are welcome! 😉 

Seriously though, we'll need a little more information to track this down. If you can find the problem and fix it yourself we welcome the contribution. How do you know the memory is leaking? How many iterations of updates have you tried and is the leak continuously leaking? What is `line` in your example above? The `LinePlot` object? The `LineVisual` underneath?

**Comment by UUSim** [comment]
To elabolate, here is the complete code:
```
#!/usr/bin/env python3
"""
Plot data with different styles
"""
import numpy as np
import time
from vispy import plot as vp
  
fig = vp.Fig(size=(800, 400))

N_SAMPLES = 1000000
x = np.linspace(0, 1, N_SAMPLES)
y = np.random.rand(N_SAMPLES)-.5
line = fig[0, 0].plot((x, y))
 
print("line: {}".format(line)) # Shows that line is a vispy.scene.visuals.LinePlot

if __name__ == '__main__':
    fig.show()#run=True) # disabled to allow an update loop below
    
    lastupdate = time.time()
    while True:
        delta = time.time() - lastupdate
        if delta > 2: # Update the data every few seconds
            lastupdate = time.time()
            line.set_data((x,y), symbol=None) # if symbol=None is omitted, symbols will appear after the set_data call.
        fig.app.process_events() # Added this call to update the gui and handle mouse events
```

I have observed the memory leak by running `top`, and watching the `python3` process. Every two seconds, the memory usage increases with approximately 57MiB.

**Comment by davidh-ssec** [comment]
What `app` (backend) are you using?

**Comment by UUSim** [comment]
PyQt5 on Qt5.6.
`print("app: {}".format(fig.app))` yields `app: <Vispy app, wrapping the PyQt5 GUI toolkit>`

**Comment by ThomasXBMC** [comment]
I'm seeing the same problem (with PyQt5 on windows). However the leak only occurs if  `symbol=None` is used in the set_data() call in the loop. As a workaround I use 
`line.set_data((x,y), marker_size=0)`
No symbols and no more memory leak ....

**Comment by djhoese** [comment]
Fixed by #1488
